exec/jdbc/src/main/java/org/apache/drill/jdbc/impl/DrillDatabaseMetaDataImpl.java

```diff
@@ -184,6 +184,17 @@ private ServerMeta getServerMeta() throws SQLException {
     return serverMeta;
   }
 
+  /**
+   * The same as {@link DrillDatabaseMetaDataImpl#getServerMeta()} but updates server metadata for the every call
+   * @return server meta information
+   * @throws SQLException for error when getting server meta
+   */
+  private ServerMeta getUpdatedServerMeta() throws SQLException {
```

Comment:
if you set quoting_identifier in the connection string, it's part of the session, and metadata would return the correct value. But does it need to reflect it if user set it after the connection is established and used? 
Also, it seems the JDBC driver for SQLServer always return the default quoting: https://github.com/Microsoft/mssql-jdbc/blob/master/src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDatabaseMetaData.java#L836